bin/completed_segments.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+""" Find (and delete) fully completed kafka-backup segments """
+
+import os
+import re
+import sys
+import argparse
+
+
+def filename_pattern():
+    """ Kafka-backup segment/index filename pattern """
+    return 'segment_partition_%s_from_offset_%s_%s'
+
+
+def parse_args():
+    """ Parse cmdline args """
+    parser = argparse.ArgumentParser(
+        description='Find (and delete) fully completed kafka-backup segments'
+    )
+    parser.add_argument(
+        '-d', '--delete',
+        help='Delete fully completed segment files',
+        action='store_true',
+        default=False,
+    )
+    parser.add_argument(
+        'target_dir',
+        help='Backup directory (target.dir)',
+        default=os.getcwd(),
+    )
+    return parser.parse_args()
+
+
+def collect_segments(target_dir):
+    """ Collect segment partitions & offsets per directory (topic) """
+    # Make regex from pattern
+    fregex = re.compile(r"^%s$" % (filename_pattern() % (r'(\d{3})', r'(\d{10})', 'records')))
+    res = {}
+    # Traverse dirtree to collect offsets in partitions per topic
+    for tdir, _, files in os.walk(target_dir):
+        if tdir == target_dir:
+            continue
+        res[tdir] = {}
+        for segfile in sorted(files):
+            match = fregex.match(segfile)
+            if match:
+                partition = match.group(1)
+                offset = match.group(2)
+                if partition not in res[tdir]:
+                    res[tdir][partition] = []
+                res[tdir][partition].append(offset)
+    return res
+
+
+def main():
+    """ int main(int argc, char **argv) """
+    args = parse_args()
+
+    segs = collect_segments(args.target_dir)
+
+    for tdir, seg_data in segs.items():
+        for partition, offsets in seg_data.items():
+            for i, offset in enumerate(offsets):
+                # Output file path if this file is not last one in current partition
+                if i < len(offsets) - 1:
+                    index_file = os.path.join(tdir, filename_pattern() % (partition, offset, 'index'))
+                    records_file = os.path.join(tdir, filename_pattern() % (partition, offset, 'records'))
+                    print(index_file)
```

Comment:
maybe print something like:
`topic ABC, First offset 123 – Index file: … Records File: …`

Reply:
I'd prefer to remove print here tbh.. though it's useful in logs. I have this script running as systemd timer so I have this in journald. Not sure if anyone need that level of output you're talking about. Plain list of files is good from another point. You can use `| xargs` to process it on your own e.g.

Reply:
So then we have to modes:

* `--list` to show the completed segments
* `--delet` to delete them ;)

Reply:
but what about your requested format? Do you want me to implement it as well?

Reply:
Hmm... btw, what to do if no cmdline option is given then? :-D

Reply:
If no option is given show the help message ;)
regarding the format: I think the `--list` will be used by humans, so I would prefer something human-readible – e.g. the requested format :)

Reply:
I thought on this.. It seems here is sysadmins POV is intersecting with developers POV :-D I see no reason to have complex format as I only interested in further processing of those files. With 'human' format I'll need to do special ('advanced') parsing to split line into index & records file. Let's say I'd like to use `shred` on files instead of just unlinking them..

So from my POV we may:
1) leave it as is (print by default, delete if requested)
2) add requested (human-readable) output as default behaviour, then add `-l|--list` (or `-p|--print`) options to just output filenames.

What do you think?

Reply:
sounds good to me :)

Reply:
ok, so I'll implement 2nd case then

Reply:
So after this change default output looks like this:
```
Topic mssl_historicalconsumption, First offset 0002020280 - Index file: segment_partition_000_from_offset_0002020280_index Records File: segment_partition_000_from_offset_0002020280_records
Topic mssl_historicalconsumption, First offset 0002136707 - Index file: segment_partition_000_from_offset_0002136707_index Records File: segment_partition_000_from_offset_0002136707_records
Topic mssl_historicalconsumption, First offset 0002068770 - Index file: segment_partition_001_from_offset_0002068770_index Records File: segment_partition_001_from_offset_0002068770_records
Topic mssl_historicalconsumption, First offset 0002019415 - Index file: segment_partition_002_from_offset_0002019415_index Records File: segment_partition_002_from_offset_0002019415_records
Topic mssl_historicalconsumption, First offset 0002135529 - Index file: segment_partition_002_from_offset_0002135529_index Records File: segment_partition_002_from_offset_0002135529_records
Topic mssl_historicalconsumption, First offset 0002018055 - Index file: segment_partition_003_from_offset_0002018055_index Records File: segment_partition_003_from_offset_0002018055_records
Topic mssl_historicalconsumption, First offset 0002134658 - Index file: segment_partition_003_from_offset_0002134658_index Records File: segment_partition_003_from_offset_0002134658_records
Topic mssl_historicalconsumption, First offset 0002014747 - Index file: segment_partition_004_from_offset_0002014747_index Records File: segment_partition_004_from_offset_0002014747_records
Topic mssl_historicalconsumption, First offset 0002131629 - Index file: segment_partition_004_from_offset_0002131629_index Records File: segment_partition_004_from_offset_0002131629_records
Topic ebs_historicalconsumption, First offset 0004475498 - Index file: segment_partition_000_from_offset_0004475498_index Records File: segment_partition_000_from_offset_0004475498_records
Topic ebs_historicalconsumption, First offset 0004509993 - Index file: segment_partition_001_from_offset_0004509993_index Records File: segment_partition_001_from_offset_0004509993_records
Topic ebs_historicalconsumption, First offset 0004480016 - Index file: segment_partition_002_from_offset_0004480016_index Records File: segment_partition_002_from_offset_0004480016_records
Topic ebs_historicalconsumption, First offset 0004481663 - Index file: segment_partition_003_from_offset_0004481663_index Records File: segment_partition_003_from_offset_0004481663_records
Topic ebs_historicalconsumption, First offset 0004495759 - Index file: segment_partition_004_from_offset_0004495759_index Records File: segment_partition_004_from_offset_0004495759_records
```

Current index & record (last one) is not shown here. Just to be sure, is this what you expected?

Reply:
Looks good to me ;)

Reply:
Done! Pls have a look